realtime_tools/include/realtime_tools/realtime_publisher.hpp

```diff
@@ -116,14 +124,14 @@ class RealtimePublisher
   /**
   * \brief Try to acquire the data lock for non-realtime message publishing
   *
-  * It first checks if the current state allows non-realtime message publishing (turn_ == NON_REALTIME)
+  * It first checks if the current state allows non-realtime message publishing (turn_ == REALTIME)
   * and then attempts to lock
   *
   * \return true if the lock was successfully acquired, false otherwise
   */
   bool trylock()
   {
-    if (turn_.load(std::memory_order_acquire) == State::NON_REALTIME && msg_mutex_.try_lock()) {
+    if (turn_.load(std::memory_order_acquire) == State::REALTIME && msg_mutex_.try_lock()) {
```

Comment:
I feel like turn_ should be checked under mutex protection. So I would think the order here should be reversed. Although if the turn check fails you will need to release lock

Reply:
Oh yes, I forgot. I think you are right that it shouldn't be an issue.